packages/consumption/src/modules/requests/itemProcessors/deleteAttribute/DeleteAttributeRequestItemProcessor.ts

```diff
@@ -0,0 +1,121 @@
+        const attribute = await this.consumptionController.attributes.getLocalAttribute(requestItem.attributeId);
+
+        if (!attribute) {
+            return ValidationResult.error(
+                CoreErrors.requests.invalidRequestItem(`The Attribute with the given attributeId '${requestItem.attributeId.toString()}' could not be found.`)
+            );
+        }
+
+        if (!attribute.isOwnSharedAttribute(this.accountController.identity.address)) {
+            return ValidationResult.error(
+                CoreErrors.requests.invalidRequestItem(`The Attribute with the given attributeId '${requestItem.attributeId.toString()}' is not an own shared Attribute.`)
+            );
+        }
+
+        if (attribute.deletionInfo?.deletionStatus === DeletionStatus.DeletedByPeer) {
+            return ValidationResult.error(CoreErrors.requests.invalidRequestItem("The Attribute was already deleted by the peer."));
+        }
+
+        if (attribute.deletionInfo?.deletionStatus === DeletionStatus.ToBeDeletedByPeer) {
+            return ValidationResult.error(CoreErrors.requests.invalidRequestItem("The peer already accepted the deletion of the Attribute."));
+        }
+
+        if (attribute.content.owner.equals(recipient)) {
+            return ValidationResult.error(CoreErrors.requests.invalidRequestItem("The deletion of an own Attribute doesn't need to be requested."));
+        }
+
+        if (!attribute.shareInfo.peer.equals(recipient)) {
+            return ValidationResult.error(
+                CoreErrors.requests.invalidRequestItem("The deletion of a shared Attribute can only be requested from the peer the Attribute is shared with.")
+            );
+        }
+
+        return ValidationResult.success();
+    }
+
+    public override canAccept(_requestItem: DeleteAttributeRequestItem, params: AcceptDeleteAttributeRequestItemParametersJSON, _requestInfo: LocalRequestInfo): ValidationResult {
+        const deletionDate = CoreDate.from(params.deletionDate);
+
+        if (!deletionDate.dateTime.isValid) {
+            return ValidationResult.error(CoreErrors.requests.invalidAcceptParameters("The deletionDate is invalid."));
+        }
+
+        if (deletionDate.isBefore(CoreDate.utc())) {
+            return ValidationResult.error(CoreErrors.requests.invalidAcceptParameters("The deletionDate must be located in the future."));
+        }
+
+        return ValidationResult.success();
+    }
+
+    public override async accept(
+        requestItem: DeleteAttributeRequestItem,
+        params: AcceptDeleteAttributeRequestItemParametersJSON,
+        _requestInfo: LocalRequestInfo
+    ): Promise<DeleteAttributeAcceptResponseItem> {
+        const attribute = await this.consumptionController.attributes.getLocalAttribute(requestItem.attributeId);
+        if (typeof attribute === "undefined") {
```

Comment:
I was told the more precise `if (typeof attribute === "undefined")` is preferred, since it doesn't allow `attribute` to be `false` or `null`. Not sure if anyone has a strong opinion on this.